src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -38,6 +38,10 @@ internal class DurableOrchestrationContext : DurableCommonContext, IDurableOrche
         private readonly Dictionary<string, Queue<string>> bufferedExternalEvents =
             new Dictionary<string, Queue<string>>(StringComparer.OrdinalIgnoreCase);
 
+        private readonly int maxActionCount;
+
+        private static int actionCount;
```

Comment:
Remove `static`. Any code that we add to the `DurableOrchestrationContext` class that can be invoked from an orchestrator function must follow all the determinism rules, and static variables violate those rules. I think the biggest problem in this case would be that multiple orchestrations running concurrently would be updating the same state and cause the exception to be thrown when it shouldn't be.